test/parallel/test-error-codes-impl-doc.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+// Flags: --expose-internals
+
+// Note: this script reads Node.js sources and docs, and expects to be run on
+// the very same Node.js version as the source and doc tree is.
+
+require('../common');
+const assert = require('assert');
+const errors = require('internal/errors').codes;
+const fs = require('fs');
+const natives = process.binding('natives');
+
+// --report prints only the list of failed checks and does not assert
+const report = process.argv[2] === '--report';
+const results = [];
+function check(ok, type, name, reason) {
+  if (report) {
+    if (!ok) results.push(`${type}: ${name} - ${reason}`);
+  } else {
+    assert.ok(ok, `${type}: ${name} - ${reason}`);
+  }
+}
+
+const set = {
+  all: new Set(),
+  js: new Set(),
+  doc: new Set(),
+  cpp: new Set(),
+  noargs: new Set(),
+  // `set.manual` contains errors manually created from js and not defined in
+  // internals/errors. That is not scanned now, update this list if required.
+  manual: new Set(['ERR_HTTP2_ERROR', 'ERR_UNKNOWN_BUILTIN_MODULE'])
+};
+
+// File containing error definitions
+const jsdata = natives['internal/errors'];
+// File containing error documentation
+const docdata = fs.readFileSync('doc/api/errors.md', 'utf-8');
```

Comment:
Can you make this an absolute path so that it can be run somewhere other than the project directory?

Reply:
Fixed, thanks @joyeecheung!